src/index.js

```diff
@@ -5,6 +5,12 @@ const deviceList = require("./device_list.js");
 const idbCompanion = require("./idb_companion.js");
 const process = require("process");
 
+/**
+ * A utility function to introduce a delay.
+ * @param ms - The delay in milliseconds.
+ */
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
```

Comment:
just noticed, these are JavaScript files and not Typescript, so `: number` doesn't apply here.


```suggestion
const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
```

Reply:
Also, you can use `util.promisify` to accomplish that.

```suggestion
const util = require("util");

const delay = util.promisify(setTimeout);
```